edx_event_bus_kafka/internal/consumer.py

```diff
@@ -117,39 +118,27 @@ def __init__(self, topic, group_id, signal):
         self.consumer = self._create_consumer()
         self._shut_down_loop = False
 
-    # return type (Optional[DeserializingConsumer]) removed from signature to avoid error on import
+    # return type (Optional[Consumer]) removed from signature to avoid error on import
```

Comment:
Looks like we should drop the Optional (not a result of this PR, though).

Reply:
accidentally took out the bit that will return none if confluent-kafka isn't available. will fix.

Reply:
It looks to me like it already couldn't return None, so I don't know that there's any need to change that.

Reply:
It would make it more consistent with the producer

Reply:
I'm not sure we want to be consistent in that direction. It just delays the error until consume/produce time. I suspect the Optional part is a holdover from an earlier design where we didn't have a wrapper around the consumer object. Not sure on that, though.

Reply:
(Specifically, we'll get "can't call method poll() on type None" or some other less-useful error.)

Reply:
Updated since I don't think it's worth delaying the PR. We may want to reconsider when we make the consumer more configurable like the producer.